contracts/src/v0.8/EIP_712_1014_4337/EIP_712_1014_4337.t.sol

```diff
@@ -0,0 +1,270 @@
+        assertEq("", greeter.getGreeting());
+
+        // Use existing entry point contract.
+        entryPoint = EntryPoint(payable(ENTRY_POINT));
+    }
+
+    /// @dev Test case for user that already has a Smart Contract Account.
+    /// @dev EntryPoint.sol should use the existing SCA to execute the meta transaction.
+    function testEIP712EIP4337WithExistingSmartContractAccount() public {
+        // Use forked Create2Factory contract.
+        Create2Factory factory = Create2Factory(CREATE2_FACTORY);
+
+        // Pre-calculate user smart contract account address.
+        bytes32 salt = bytes32(uint256(uint160(LINK_WHALE)) << 96);
+        bytes memory fullInitializeCode = bytes.concat(
+            initailizeCode,
+            abi.encode(LINK_WHALE)
+        );
+        bytes32 initializeCodeHash = keccak256(fullInitializeCode);
+        address toDeployAddress = factory.findCreate2Address(
+            salt,
+            initializeCodeHash
+        );
+
+        // Deploy the end-contract.
+        changePrank(SENDER_CREATOR);
+        factory.callCreate2(salt, fullInitializeCode);
+        changePrank(LINK_WHALE);
+
+        // Ensure a correct deployment and a functioning end-contract.
+        address deployedAddress = factory.findCreate2Address(
+            salt,
+            initializeCodeHash
+        );
+        assertEq(deployedAddress, address(0));
+        assertEq(true, SCA(toDeployAddress).isSCA());
+        assertEq(LINK_WHALE, SCA(toDeployAddress).s_owner());
+
+        // Create the calldata for a setGreeting call.
+        string memory greeting = "hi";
+        bytes memory encodedGreetingCall = abi.encodeWithSelector(
+            Greeter.setGreeting.selector,
+            greeting
+        );
+
+        // For developers: log the final hash of the SCA call to easily produce a signature off-chain.
+        bytes memory fullEncoding = abi.encodeWithSelector(
+            SCA.executeTransaction.selector,
+            address(greeter),
+            encodedGreetingCall
+        );
+        bytes32 hashOfEncoding = keccak256(
+            abi.encode(
+                toDeployAddress,
+                fullEncoding,
+                LINK_WHALE,
+                uint256(1), // nonce++
+                block.chainid
+            )
+        );
+        console.logBytes32(hashOfEncoding);
+
+        // Construct the user operation.
+        UserOperation memory op = UserOperation({
+            sender: toDeployAddress,
+            nonce: 0,
+            initCode: "",
+            callData: fullEncoding,
+            callGasLimit: 1_000_000,
+            verificationGasLimit: 1_000_000,
+            preVerificationGas: 1_000_000,
+            maxFeePerGas: 100,
+            maxPriorityFeePerGas: 200,
+            paymasterAndData: "",
+            signature: signature
+        });
+
+        // Deposit funds for the transaction.
+        entryPoint.depositTo{value: 10 ether}(toDeployAddress);
```

Comment:
Who calls `depositTo` in practice? Is it the end user or the bundler? Or someone else?

Reply:
Aggregators and paymasters may be different yes